prometheus-export/src/main/scala/StreamingQuerySource.scala

```diff
@@ -80,6 +90,41 @@ object StreamingQuerySource extends Logging {
     }
   }
 
+  // An iceberg structured streaming offset looks like this
+  // {
+  //   "description" : "org.apache.iceberg.spark.source.SparkMicroBatchStream@32c41681",
+  //   "startOffset" : {
+  //     "version" : 1,
+  //     "snapshot_id" : 3202691127021887425,
+  //     "position" : 1,
+  //     "scan_all_files" : false
+  //   },
+  //   "endOffset" : {
+  //     "version" : 1,
+  //     "snapshot_id" : 1926525026331941549,
+  //     "position" : 1,
+  //     "scan_all_files" : false
+  //   },
+  //   "latestOffset" : null,
+  //   "numInputRows" : 15,
+  //   "inputRowsPerSecond" : 0.25,
+  //   "processedRowsPerSecond" : 5.210142410559222
+  // }
+  def icebergOffset(offsetJSON: String): Map[String, Either[Long, Boolean]] = {
+    if (offsetJSON == null || offsetJSON == "") {
+      return Map()
+    }
+
+    offsetJSON.trim.headOption match {
+      case Some('{') if offsetJSON.trim.lastOption.contains('}') =>
+        implicit val formats = DefaultFormats
+        parse(offsetJSON).extract[Map[String, Either[Long, Boolean]]]
+      case _ =>
+        log.trace(s"Streaming offset data is not processable: $offsetJSON")
+        Map()
+    }
```

Comment:
Similarly, you could change this here to avoid `return`.

```suggestion
    if (offsetJSON != null && offsetJSON.nonEmpty) {
      val trimmedOffsetJSON = offsetJSON.trim
      trimmedOffsetJSON.headOption match {
        case Some('{') if trimmedOffsetJSON.lastOption.contains('}') =>
          implicit val formats = DefaultFormats
          parse(offsetJSON).extract[Map[String, Either[Long, Boolean]]]
        case _ =>
          log.trace(s"Streaming offset data is not processable: $offsetJSON")
          Map()
      }
    } else {
      Map()
```